Recruit-Training/Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -16,23 +22,72 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+
+    return totalAmphoursUsed_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
+    if(current_ < 0){
+        return true;
+    }
     return false;
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+    double totalHoursRemaining, msRemaining;
+        int h, m, s, ms;
+
+        if(isCharging())
```

Comment:
In general, even if what follows is just one line, we should still encapsulate the statement in brackets. That way in the future, people can add stuff to the statement as needed without accidentally causing bugs.

![image](https://user-images.githubusercontent.com/15304203/71850678-155d0e80-3092-11ea-97c0-77da1ed1e5a2.png)

^ Apple's goto fail is an example of that (also never use goto!)

Propagate this to your other if/else statements and loops